### PR TITLE
feat(helmfile): Parse multidoc yaml

### LINKED ISSUE #8682: [HelmFile] Multidoc YAML is not supported
**What Renovate type, platform and version are you using?**

GitHub. Self-hosted "renovateVersion": "24.49.7"

**Describe the bug**

Helmfile allows you to use multidoc YAML. Renovate crash when parsing multidoc YAML.

**Relevant debug logs**

```
DEBUG: Failed to parse helmfile helmfile.yaml (repository=Insly/aws-configuration)
       "err": {
         "name": "YAMLException",
         "reason": "expected a single document in the stream, but found more",
         "message": "expected a single document in the stream, but found more",
         "stack": "YAMLException: expected a single document in the stream, but found more\n    at load (/usr/local/lib/node_modules/renovate/node_modules/js-yaml/lib/js-yaml/loader.js:1622:9)\n    at Object.safeLoad (/usr/local/lib/node_modules/renovate/node_modules/js-yaml/lib/js-yaml/loader.js:1637:10)\n    at Object.extractPackageFile (/usr/local/lib/node_modules/renovate/dist/manager/helmfile/extract.js:37:33)\n    at Object.extractPackageFile (/usr/local/lib/node_modules/renovate/dist/manager/index.js:64:13)\n    at Object.getManagerPackageFiles (/usr/local/lib/node_modules/renovate/dist/workers/repository/extract/manager-files.js:41:41)\n    at async /usr/local/lib/node_modules/renovate/dist/workers/repository/extract/index.js:42:30\n    at async Promise.all (index 2)\n    at async Object.extractAllDependencies (/usr/local/lib/node_modules/renovate/dist/workers/repository/extract/index.js:41:28)\n    at async Object.extract (/usr/local/lib/node_modules/renovate/dist/workers/repository/process/extract-update.js:75:24)\n    at async Object.extractDependencies (/usr/local/lib/node_modules/renovate/dist/workers/repository/process/index.js:60:41)\n    at async Object.renovateRepository (/usr/local/lib/node_modules/renovate/dist/workers/repository/index.js:64:56)\n    at async Object.start (/usr/local/lib/node_modules/renovate/dist/workers/global/index.js:77:13)\n    at async /usr/local/lib/node_modules/renovate/dist/renovate.js:28:24"
       },
       "fileName": "kubernetes/helmfile.yaml"
```

**Have you created a minimal reproduction repository?**

https://github.com/kuzaxak/renovate-helmfile-reproductions

**Additional context**

Renovate Bot should parse and analyze each doc in YAML file.

### PR DESCRIPTION
**Pull request by kuzaxak**
We should skip empty docs to be able to parse last one with contains
releases map.

## Changes:

* Skip empty docs in multidoc yaml
* Return `null` in case of empty release map

## Context:

Closes #8682

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [X] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/helmfile/extract.ts [hunk 1]
```diff
@@ -35,8 +35,7 @@ export function extractPackageFile(
   }
   for (const doc of docs) {
     if (!(doc && is.array(doc.releases))) {
-      logger.debug({ fileName }, 'helmfile.yaml has no releases');
-      return null;
+      continue;
```
Comment (rarkins) [inline lib/manager/helmfile/extract.ts:38]:
```suggestion
      continue; // eslint-disable-line no-continue
```

File: lib/manager/helmfile/extract.ts [hunk 2]
```diff
@@ -94,5 +93,10 @@ export function extractPackageFile(
     });
   }
 
+  if (deps.length == 0) {
```
Comment (rarkins) [inline lib/manager/helmfile/extract.ts:96]:
```suggestion
  if (!deps.length) {
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Let's make sure we have a test

**Comment by kuzaxak** [comment]
> Let's make sure we have a test

What do you mean by that? 
Unit tests?

Could you add a link to example, I can help you with that. We really need this functionality :)

**Comment by rarkins** [comment]
There are already unit tests for this manager in the same directory - just none with multi doc. Check out the corresponding spec.js file

**Pull request closed without merging**